src/us/kbase/workspace/test/database/provenance/CreditMetadataTest.java

```diff
@@ -0,0 +1,1052 @@
+			try {
+				ResourceType.getResourceType(rt);
+				fail(EXP_EXC);
+			} catch (Exception got) {
+				TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+						"Invalid resourceType: " + rt));
+			}
+		}
+		for (final String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
+			try {
+				ResourceType.getResourceType(nullOrWs);
+				fail(EXP_EXC);
+			} catch (Exception got) {
+				TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+						RESOURCE_TYPE_NON_WS_NULL));
+			}
+		}
+	}
+
+	/**
+	 * Check the CreditMetadata has the expected fields
+	 *
+	 * @param cm
+	 *                instance to check
+	 * @param identifier
+	 *                expected resource ID
+	 * @param rt
+	 *                expected ResourceType
+	 * @param contributors
+	 *                list of Contributor objects expected to be returned by
+	 *                `getContributors`;
+	 * @param titles
+	 *                list of Title objects expected to be returned by
+	 *                `getTitles`;
+	 * @param expectedMap
+	 *                key/value pairs are the field names and values for the string
+	 *                fields that are expected not to be null
+	 * @param comments
+	 *                list of strings expected to be returned by `getComments`;
+	 * @param dates
+	 *                list of EventDates expected to be returned by
+	 *                `getDates`;
+	 *                if null, should return an empty list
+	 * @param funding
+	 *                list of FundingReferences expected to be returned by
+	 *                `getFunding`;
+	 *                if null, should return an empty list
+	 * @param relatedIdentifiers
+	 *                list of PermanentIDs expected to be returned by
+	 *                `getRelatedIdentifiers`;
+	 *                if null, should return an empty list
+	 */
+	private void assertCreditMetadataFields(final CreditMetadata cm,
+			final String identifier,
+			final ResourceType rt,
+			final List<Contributor> contributors,
+			final List<Title> titles,
+			final String license,
+			final String version,
+			final List<String> comments,
+			final List<EventDate> dates,
+			final List<FundingReference> funding,
+			final List<PermanentID> relatedIdentifiers) {
+
+		assertThat(INCORRECT_ID, cm.getIdentifier(), is(identifier));
+		assertThat(INCORRECT_TYPE, cm.getResourceType(), is(rt));
+		assertThat(INCORRECT_COMMENTS, cm.getComments(),
+				is(comments));
+		assertThat(INCORRECT_CONTRIBUTORS, cm.getContributors(),
+				is(contributors));
+		assertThat(INCORRECT_DATES, cm.getDates(),
+				is(dates));
+		assertThat(INCORRECT_FUNDING, cm.getFunding(),
+				is(funding));
+		assertThat(INCORRECT_RELATED_IDS, cm.getRelatedIdentifiers(),
+				is(relatedIdentifiers));
+		assertThat(INCORRECT_TITLES, cm.getTitles(),
+				is(titles));
+		assertThat(INCORRECT_LICENSE, cm.getLicense(), is(Optional.ofNullable(license)));
+		assertThat(INCORRECT_VERSION, cm.getVersion(), is(Optional.ofNullable(version)));
```

Comment:
These last two lines are recapitulating the code under test with the `ofNullable` stuff. Better to make the incoming type optional

Reply:
:+1: